### PR TITLE
Fixes for integration and sanity test cases 

### PR DESCRIPTION
**Pull request by mukhoakash**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: pkg/provisioner/crdprovisioner.go [hunk 1]
```diff
@@ -196,6 +196,10 @@ func (c *CrdProvisioner) CreateVolume(
 					}
 				}
 			} else {
+				// If current request has different specifications than the existing volume, return error.
+				if updateAzVolumeSpec(azVolumeInstance, maxMountReplicaCount, capacityRange, parameters, secrets, volumeContentSource, accessibilityReq) {
```
Comment (mukhoakash) [inline pkg/provisioner/crdprovisioner.go:200]: Need to update the function name

**Review by edreed — APPROVED** [review]

File: test/latest/azuredisk-csi-driver/templates/disk.csi.azure.com_azvolumeattachments.yaml [hunk 2]
```diff
@@ -85,11 +85,39 @@ spec:
               error:
                 description: Error occured during attach/detach of volume
                 properties:
+                  currentNode:
```
Comment (sunpa93) [inline test/latest/azuredisk-csi-driver/templates/disk.csi.azure.com_azvolumeattachments.yaml:88]: Where are these changes from? It seems like changes from `v1alpha1/types.go` were not added to the commit. And are these fields used anywhere by the crdprovisioner or the controllers?

File: test/latest/azuredisk-csi-driver/templates/disk.csi.azure.com_azvolumes.yaml [hunk 3]
```diff
@@ -149,18 +149,46 @@ spec:
               error:
                 description: Error occured during creation/deletion of volume
                 properties:
+                  currentNode:
```
Comment (sunpa93) [inline test/latest/azuredisk-csi-driver/templates/disk.csi.azure.com_azvolumes.yaml:152]: Same question as above 👍

File: test/latest/azuredisk-csi-driver/templates/disk.csi.azure.com_azvolumeattachments.yaml (hunk 2 above)
Comment (mukhoakash) [reply test/latest/azuredisk-csi-driver/templates/disk.csi.azure.com_azvolumeattachments.yaml:88]: https://github.com/abhisheksinghbaghel/azuredisk-csi-driver/blob/73f336b4a0cb7e4a279593ed790049b5cccacc29/pkg/apis/azuredisk/v1alpha1/types.go#L181

Looks like this line bringing in the change.

Comment (mukhoakash) [reply test/latest/azuredisk-csi-driver/templates/disk.csi.azure.com_azvolumeattachments.yaml:88]: Yes, its being used in the provisioners to handle the DanglingAttachError.

File: pkg/controller/azvolume.go [hunk 4]
```diff
@@ -288,6 +297,13 @@ func (r *reconcileAzVolume) CleanUpAzVolumeAttachment(ctx context.Context, azVol
 	}
 
 	volRequirement, _ := labels.NewRequirement(VolumeNameLabel, selection.Equals, []string{azVolume.Spec.UnderlyingVolume})
```
Comment (nearora-msft) [inline pkg/controller/azvolume.go:299]: Should it volRequirement, err?

File: pkg/controller/azvolumeattachment.go [hunk 5]
```diff
@@ -762,6 +786,13 @@ func (r *reconcileAzVolumeAttachment) CleanUpAzVolumeAttachment(ctx context.Cont
 	}
 
 	volRequirement, _ := labels.NewRequirement(VolumeNameLabel, selection.Equals, []string{azVolume.Spec.UnderlyingVolume})
```
Comment (nearora-msft) [inline pkg/controller/azvolumeattachment.go:788]: Same as above. Should it volRequirement, err?

Comment (mukhoakash) [reply pkg/controller/azvolumeattachment.go]: Fixed

File: pkg/provisioner/crdprovisioner.go [hunk 6]
```diff
@@ -196,6 +196,10 @@ func (c *CrdProvisioner) CreateVolume(
 					}
 				}
 			} else {
+				// If current request has different specifications than the existing volume, return error.
+				if !isAzVolumeSpecSameAsRequestParams(azVolumeInstance, maxMountReplicaCount, capacityRange, parameters, secrets, volumeContentSource, accessibilityReq) {
```
[the remark this answers is not in the document]
Comment (mukhoakash) [reply pkg/provisioner/crdprovisioner.go:200]: Well. there is a scenario where if the volume already exists and the current request has different params, an error should be returned. There is a test case in sanity test related to this as well.

When the same description request is sent in the request, the details of the existing volume (without any error) should be returned.

**Review by nearora-msft — APPROVED** [review]

**Pull request merged**